app/scripts/modules/core/securityGroup/securityGroupWriter.service.ts

```diff
@@ -0,0 +1,70 @@
+import {assignWith, isUndefined} from 'lodash';
+import {module} from 'angular';
+
+import {TASK_EXECUTOR, TaskExecutor, IJob} from 'core/task/taskExecutor';
+import {INFRASTRUCTURE_CACHE_SERVICE, InfrastructureCacheService} from 'core/cache/infrastructureCaches.service';
+import {ISecurityGroup} from 'core/domain';
+import {Application} from 'core/application/application.model';
+import {ITask} from 'core/task/task.read.service';
+
+export interface ISecurityGroupJob extends IJob {
+  credentials?: string;
+  regions?: string[];
+  securityGroupName?: string;
+}
+export class SecurityGroupWriterService {
+
+  static get $inject(): string[] {
+    return ['infrastructureCaches', 'taskExecutor'];
+  }
+
+  constructor(private caches: InfrastructureCacheService,
+              private executor: TaskExecutor) {}
+
+  public deleteSecurityGroup(securityGroup: ISecurityGroup,
+                             application: Application,
+                             params: ISecurityGroupJob): ng.IPromise<ITask> {
+
+    params.type = 'deleteSecurityGroup';
+    params.securityGroupName = securityGroup.name;
+    params.regions = [securityGroup.region];
+    params.credentials = securityGroup.accountId;
+
+    const operation: ng.IPromise<ITask> = this.executor.executeTask({
+      job: [params],
+      application: application,
+      description: `Delete Security Group: ${securityGroup.name}`
+    });
+    this.caches.clearCache('securityGroups');
+
+    return operation;
+  }
+
+  public upsertSecurityGroup(securityGroup: ISecurityGroup,
+                             application: Application,
+                             description: string,
+                             params: ISecurityGroupJob): ng.IPromise<ITask> {
+
+    params.type = 'upsertSecurityGroup';
+    params.credentials = securityGroup.credentials || securityGroup.accountName;
+
+    // We want to extend params with all attributes from command, but only if they don't already exist.
+    assignWith(params, securityGroup, (value: any, other: any) => {
+      return isUndefined(value) ? other : value;
+    });
```

Comment:
I looked through too and agree - I couldn't find any cases where it would be explicitly set so I would agree we can remove it - you have far better a feel for the codebase so I'm somewhat reticent to make certain assumptions.  I also agree we should defer to native implementations where possible.  I'll convert it over.